core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -864,15 +925,37 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
     }
 
     /**
+     * Deletes a snapshot from the repository, looking up the {@link Snapshot} reference before deleting.
+     * If the snapshot is still running cancels the snapshot first and then deletes it from the repository.
```

Comment:
that's not whats happening here. If a snapshot with the name exists in the repository only the snapshot in the repository is deleted. The running snapshot is not canceled. We should do both imo.